### ISSUE TITLE
Queue full and master branch failing

### ISSUE THREAD

**Issue by cachedout** [elastic/hey-apm#196]
Hi @axw 

We appear to be [seeing a continual failure to complete the Hey APM Benchmark test suite within the allotted time period of 1hr](https://apm-ci.elastic.co/job/apm-server/job/apm-hey-test-benchmark/).

The log [containing the errors can be found here](https://apm-ci.elastic.co/blue/rest/organizations/jenkins/pipelines/apm-server/pipelines/apm-hey-test-benchmark/runs/769/nodes/91/steps/163/log/?start=0).

It appears as if the server is returning a `queue full` message.

Example: `[2021-08-02T05:10:38.947Z] hey-apm_1          | 2021/08/02 05:10:38 logger.go:10: [debug] request failed: request failed with 503 Service Unavailable: {"accepted":0,"errors":[{"message":"queue is full"}]} (next request in ~0s)`

**Comment by axw** [comment]
@cachedout do you know if we capture the apm-server log? That log is just for the hey-apm process, and doesn't provide any clues as to why the queue is full. Looks like something is permanently failing and preventing events from getting indexed.

**Comment by cachedout** [comment]
@axw Absolutely! The collective logs for all the dockerized services for that job [can be found here](https://apm-ci.elastic.co/job/apm-server/job/apm-hey-test-benchmark/770/artifact/src/github.com/elastic/hey-apm/build/environment.txt/*view*/). The APM Server logs are prefixed with `apm-server_1`.

**Comment by axw** [comment]
Thanks @cachedout. Looks like we're hitting https://github.com/elastic/apm-server/issues/5807

> [36mapm-server_1       |[0m {"log.level":"error","@timestamp":"2021-08-03T05:03:13.732Z","log.logger":"pipelines","log.origin":{"file.name":"pipeline/register.go","file.line":50},"message":"Pipeline registration failed for apm_convert_destination_address.","service.name":"apm-server","event.dataset":"apm-server","ecs.version":"1.6.0"}

I was already planning to look at that soon, now I'll look at it a bit sooner ;)

**Comment by simitt** [comment]
It sounds like this should have been fixed with https://github.com/elastic/apm-server/issues/5807. Can this be closed?
